Source/WebCore/accessibility/AXTextMarker.cpp

```diff
@@ -633,6 +633,50 @@ AXTextMarker AXTextMarker::findLastBefore(std::optional<AXID> stopAtID) const
     return marker;
 }
 
+AXTextMarkerRange AXTextMarker::rangeWithSameStyle() const
+{
+    RELEASE_ASSERT(!isMainThread());
+
+    if (!isValid())
+        return { };
+
+    // FIXME: findMarker() walks index-by-index, which is wasteful in this function, since styles will only differ between objects
+    // and not individual indices, so in the future we should walk entire objects at a time to compute and compare less styles.
+
+    auto originalStyle = object()->stylesForAttributedString();
+    auto findMarkerWithDifferentStyle = [&] (AXDirection direction) {
+        AXID previousObjectID = object()->objectID();
+        auto currentMarker = *this;
+
+        while (true) {
+            auto newMarker = currentMarker.findMarker(direction);
```

Comment:
That would only stop at objects that have runs, and for this API I'm not sure that's right. e.g. for this HTML:

```html
<div>
    <div>Foo</div>
    Bar
</div>

<div id="d1" style="font-style: italic">
    <div>Baz</div>
</div>
```

Given a text marker pointing to the first character of "Foo", should the style range end at the first spot of the "Baz"-static-text text marker? Or should it stop at the "anchor" position that actually has different style, which is `#d1`? I would guess the latter, though it probably doesn't really matter that much, especially in this contrived example. And we don't represent anchor positions at this point anyways 🤷

Reply:
I definitely see your point. Though, since we really only care about when text style differs, it seems safe to use the "equivalent"/deepest position that has runs for the perf benefit? I can see an argument either way.

Reply:
OK yeah, `findObjectWithRuns` is probably the most pragmatic choice right now — fixed in latest commit!